src/lib/rules/strict-transport-security/strict-transport-security.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @fileoverview This rule checkes header strict-transport-security is sent for resources served over HTTPS, and max-age has proper  value.
+ */
+
+import { RuleContext } from '../../rule-context'; // eslint-disable-line no-unused-vars
+import { isDataURI, isHTTPS, normalizeString } from '../../utils/misc'; // eslint-disable-line no-unused-vars
+import { IAsyncHTMLElement, IResponse, IFetchEnd, IRule, IRuleBuilder } from '../../types'; // eslint-disable-line no-unused-vars
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The minimum period (in seconds) allowed for `max-age`. */
+        let minimum: number;
+
+        const loadRuleConfigs = () => {
+            minimum = (context.ruleOptions && context.ruleOptions.minimum) || 10886400;
+        };
+
+        const maxAgeLessThan = (headerValue: string, limit: number) => {
+            const maxAge = headerValue.match(/max-age=([^;]+);|max-age=([^;]+)$/).pop();
```

Comment:
@alrra You are right, I didn't know it could be uppercase. One question though - the doc says `Directive names are case-insensitive`. So that means `MAX-AGE=...` is not allowed and should **fail** the test, right? Also,  It mentioned `The order of appearance of directives is not significant`, so if we `split`, it's possible that `max-age` is not the first element. Am I understanding it correctly?